test/dd/test_state_generation.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+ * Copyright (c) 2025 Munich Quantum Software Company GmbH
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Licensed under the MIT License
+ */
+
+#include "dd/Package.hpp"
+#include "dd/StateGeneration.hpp"
+
+#include <cmath>
+#include <complex>
+#include <cstddef>
+#include <gtest/gtest.h>
+#include <memory>
+#include <vector>
+
+using namespace dd;
+
+namespace {
+double norm(const std::vector<std::complex<double>>& v) {
+  double sum{};
+  for (const auto& entry : v) {
+    sum += std::pow(std::abs(entry), 2);
```

Comment:
```suggestion
    sum += std::norm(entry);
```
There's an stdlib function for that 😉